src/polytope/models/Problem.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass, field
+from typing import List
+
+from .ProblemStatement import ProblemStatement
+from .ProblemChecker import ProblemChecker
+from .ProblemValidator import ProblemValidator
+from .ProblemTest import ProblemPublicTest, ProblemTest
+from .ProblemSolution import ProblemSolution
+
+
+@dataclass
+class Problem:
+    """! A Polytope problem dataclass."""
+
+    """! Problem ID.
+
+    * It must be auto-generated by using UUID.
+    * It must be read-only.
+
+    TODO: Define UUID type and
+          set UUID generator as a default_factory.
+    """
+    _id: str
+
+    """! Problem name.
+
+    * It must be non-empty.
+    """
+    name: str
+
+    """! Draft note for the problem."""
+    note: str
+
+    """! Time limit in milliseconds.
+
+    * It must be positive.
+    """
+    time_limit_in_ms: int
+
+    """! Memory limit in mebibytes.
+
+    * It must be positive."""
+    memory_limit_in_mib: int
+
+    """! Problem tags.
+
+    * Tags must be distinct.
+    * A tag must be non-empty.
+    """
+    tags: List[str]
```

Comment:
(comment) what do you mean by 'tags' here? Is it about problem category like CF or Solved.ac?

- Do we have provided set of tags?
- Why do we enforce non-empty tags?

Reply:
Tags can contain any information. I intended the Codeforces Polygon's tags one. For example, you can put `['KOI 2023', 'greedy', 'two-steps']` in tags.

We could also add an autocomplete function by providing frequently used tags.

`tags` can, of course, be an empty list. "A tag" of the comment means an element of `tags`. An empty tag looks nonsensical.

Reply:
I think we can clarify by restating `Each tag should be non-empty` or somehow. Thanks for your explanation!

Reply:
As you advised, I changed the comment sentence to be more clearer.

https://github.com/Studio-Polytope/Polytope/blob/940bf14f32bf3c02c87d656bdfbbcbec2db826ba/src/polytope/models/Problem.py#L45-L50